app/assets/javascripts/actions/training_modification_actions.js

```diff
@@ -0,0 +1,54 @@
+import { API_FAIL } from '../constants/api';
+import { addNotification } from '../actions/notification_actions';
+import logErrorMessage from '../utils/log_error_message';
+import request from '../utils/request';
+import { setInvalid } from './validation_actions';
+
+const performValidation = (error, dispatch) => {
+  const messages = error.responseText.message;
+  messages.forEach((message) => {
+    const lowercaseMessage = message.toLowerCase();
+    if (lowercaseMessage.includes('name')) {
+      dispatch(setInvalid('name', message));
+    } else if (lowercaseMessage.includes('slug')) {
+      dispatch(setInvalid('slug', message));
+    } else if (lowercaseMessage.includes('introduction')) {
+      dispatch(setInvalid('introduction', message));
+    } else {
+      dispatch({ type: API_FAIL, data: error });
```

Comment:
We probably should only dispatch an `API_FAIL` event only once — so I recommend switching to a `for..of` loop and doing an early return in the `else` block.

Reply:
okay

Reply:
yes, this works too. We actually just disable that rule whenever we want to use `for...of`